core/src/main/java/com/linecorp/armeria/common/DefaultFlagsProvider.java

```diff
@@ -0,0 +1,364 @@
+/*
+ *  Copyright 2022 LINE Corporation
+ *
+ *  LINE Corporation licenses this file to you under the Apache License,
+ *  version 2.0 (the "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at:
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ */
+package com.linecorp.armeria.common;
+
+import java.io.File;
+import java.net.InetAddress;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.util.Sampler;
+import com.linecorp.armeria.common.util.TransportType;
+import com.linecorp.armeria.server.TransientServiceOption;
+
+/**
+ * Implementation of {@link FlagsProvider} which provides default values to {@link Flags}.
+ */
+final class DefaultFlagsProvider implements FlagsProvider {
+
+    static final DefaultFlagsProvider INSTANCE = new DefaultFlagsProvider();
+
+    static final String VERBOSE_EXCEPTION_SAMPLER = "rate-limit=10";
+    static final int MAX_NUM_CONNECTIONS = Integer.MAX_VALUE;
+    static final int NUM_COMMON_BLOCKING_TASK_THREADS = 200; // from Tomcat default maxThreads
+    static final long DEFAULT_MAX_REQUEST_LENGTH = 10 * 1024 * 1024; // 10 MiB
+    static final long DEFAULT_MAX_RESPONSE_LENGTH = 10 * 1024 * 1024; // 10 MiB
+
+    // Use slightly greater value than the client-side default so that clients close the connection more often.
+    static final long DEFAULT_SERVER_IDLE_TIMEOUT_MILLIS = 15000; // 15 seconds
+    static final long DEFAULT_REQUEST_TIMEOUT_MILLIS = 10 * 1000; // 10 seconds
```

Comment:
nit; can we move this after `DEFAULT_RESPONSE_TIMEOUT_MILLIS ` at line 52 since the two flags are similar?

Reply:
```java
  // Use slightly greater value than the client-side default so that clients close the connection more often.
  static final long DEFAULT_SERVER_IDLE_TIMEOUT_MILLIS = 15000; // 15 seconds
  static final long DEFAULT_CLIENT_IDLE_TIMEOUT_MILLIS = 10000; // 10 seconds
  static final long DEFAULT_CONNECT_TIMEOUT_MILLIS = 3200; // 3.2 seconds
  static final long DEFAULT_WRITE_TIMEOUT_MILLIS = 1000; // 1 second

  // Use slightly greater value than the default request timeout so that clients have a higher chance of
  // getting proper 503 Service Unavailable response when server-side timeout occurs.
  static final long DEFAULT_RESPONSE_TIMEOUT_MILLIS = 15 * 1000; // 15 seconds
  static final long DEFAULT_REQUEST_TIMEOUT_MILLIS = 10 * 1000; // 10 seconds
```

like this?

Reply:
Looks good.